apps/prairielearn/src/tests/courseEditor.test.ts

```diff
@@ -403,7 +446,15 @@ function testEdit(params) {
 
   describe(`POST to ${params.url} with action ${params.action}`, function () {
     it('should load successfully', async () => {
-      const res = await fetch(params.url || locals.url, {
+      let url: string;
+      if (!params.action) {
+        const elemList = locals.$(params.formSelector);
+        assert.lengthOf(elemList, 1);
+        url = `${siteUrl}${elemList[0].attribs['action']}`;
+      } else {
+        url = params.url || locals.url;
+      }
+      const res = await fetch(url, {
```

Comment:
This is to handle the difference between POSTing to the same URL as the page you are on vs. POSTing to a different URL. I'm definitely open to a prettier way to do this.

Reply:
I think this is fine. I've reached for `@prairielearn/run` for this kind of thing before (assign to one variable based on a bunch of conditions), up to you if you like that or not.

Reply:
nice idea, done.